consensus/spos/bls/subroundSignature.go

```diff
@@ -239,3 +251,26 @@ func (sr *subroundSignature) signaturesCollected(threshold int) (bool, int) {
 
 	return n >= threshold, n
 }
+
+func (sr *subroundSignature) waitAllSignatures(ctx context.Context) {
+	remainingTime := sr.remainingTime()
+
+	select {
+	case <-ctx.Done():
+		return
+	case <-time.After(remainingTime):
+	}
+
+	select {
+	case sr.ConsensusChannel() <- true:
```

Comment:
it is not so good combining a channel and a context. Why is this needed ? the func could exit before ctx.Done or time.After because of consensusChannel() <- true

Reply:
Actually this is the channel which I need to trigger after remaining time expires and if until then leader would not received all signatures and finished subround signature meantime. This trigger will end the sub-round signature with the collected signatures no matter if they are all of them or not. So I think that I didn't need here a context approach but a little refactor to the initial implementation. I propose the below one:

Reply:
func (sr *subroundSignature) waitAllSignatures() {
	remainingTime := sr.remainingTime()
	time.Sleep(remainingTime)

	if sr.IsSubroundFinished(sr.Current()) {
		return
	}

	select {
	case sr.ConsensusChannel() <- true:
	default:
	}
}

Reply:
Just for the record: context and channels are go idiomatic. Since this function is launched on a go routine, it will end up having a context.